src/core/mac/mac.cpp

```diff
@@ -1028,10 +1028,8 @@ void Mac::BeginTransmit(void)
         // copy the frame into correct `TxFrame` for each radio type
         // (if it is not already prepared).
 
-        for (uint8_t index = 0; index < GetArrayLength(RadioTypes::kAllRadioTypes); index++)
+        for (auto radio : RadioTypes::kAllRadioTypes)
```

Comment:
How about using `RadioType` instead of `auto`?
```suggestion
        for (RadioType radio : RadioTypes::kAllRadioTypes)
```